app/crud/wh_sched.py

```diff
@@ -319,21 +321,39 @@ def regenerate_alter_statements(session: Session, schedules: List[WarehouseSched
     WarehouseAlterStatements.batch_write(session, alter_stmts, overwrite=True)
 
 
-def update_task_state(session: Session, schedules: List[WarehouseSchedules]) -> bool:
+def get_schedule_timezone(session: Session) -> timezone:
+    """
+    Fetch the 'default_timezone' from the internal.config table. If there is no timezone set or the
+    timezone which is set fails to parse, this function will return the timezone for 'America/Los_Angeles'.
+    """
+    str_tz = session.call("get_config", "default_timezone") or "America/Los_Angeles"
+    try:
+        return timezone(str_tz)
+    except UnknownTimeZoneError:
+        return timezone("America/Los_Angeles")
```

Comment:
we can't know the account timezone?

Reply:
Ah, here we can probably figured it out from Snowflake. I hadn't thought about checking the snowflake default.